src/include/btmem.h

```diff
@@ -412,8 +412,17 @@ struct __wt_page {
 	/*
 	 * Macros to copy/set the index because the name is obscured to ensure
 	 * the field isn't read multiple times.
+	 *
+	 * There are two versions of WT_INTL_INDEX_GET because the session split
+	 * generation is usually set, but it's not always required: for example,
+	 * if a page is locked for splitting, or being created or destroyed.
 	 */
-#define	WT_INTL_INDEX_COPY(page)	((page)->u.intl.__index)
+#define	WT_INTL_INDEX_GET_SAFE(page)					\
+	((page)->u.intl.__index)
+#define	WT_INTL_INDEX_GET(session, page, pindex) do {			\
```

Comment:
I agree completely -- I did it that way originally, but backed it out because there are places where we inline `WT_INTL_INDEX_GET_SAFE`, for example:
```
WT_ASSERT(session, WT_INTL_INDEX_GET_SAFE(parent) == pindex);
```
and I didn't want to turn those asserts into multiple line blocks with local variables, inside of #ifdef's.

I'll merge it as it is, if you decide you want it to be consistent let me know and I'll do it.